service/ocean/providers/aws/cluster.go

```diff
@@ -2030,3 +2031,184 @@ func (o *Filters) SetVirtualizationTypes(v []string) *Filters {
 	}
 	return o
 }
+
+func (o *LaunchSpecification) SetClusterBlockDeviceMappings(v []*ClusterBlockDeviceMapping) *LaunchSpecification {
+	if o.BlockDeviceMappings = v; o.BlockDeviceMappings == nil {
+		o.nullFields = append(o.nullFields, "BlockDeviceMappings")
+	}
+	return o
+}
+
+type ClusterBlockDeviceMapping struct {
+	DeviceName  *string     `json:"deviceName,omitempty"`
+	NoDevice    *string     `json:"noDevice,omitempty"`
+	VirtualName *string     `json:"virtualName,omitempty"`
+	EBS         *ClusterEBS `json:"ebs,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+type ClusterEBS struct {
+	DeleteOnTermination *bool                     `json:"deleteOnTermination,omitempty"`
+	Encrypted           *bool                     `json:"encrypted,omitempty"`
+	KMSKeyID            *string                   `json:"kmsKeyId,omitempty"`
+	SnapshotID          *string                   `json:"snapshotId,omitempty"`
+	VolumeType          *string                   `json:"volumeType,omitempty"`
+	IOPS                *int                      `json:"iops,omitempty"`
+	VolumeSize          *int                      `json:"volumeSize,omitempty"`
+	Throughput          *int                      `json:"throughput,omitempty"`
+	DynamicVolumeSize   *ClusterDynamicVolumeSize `json:"dynamicVolumeSize,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+type ClusterDynamicVolumeSize struct {
+	BaseSize            *int    `json:"baseSize,omitempty"`
+	SizePerResourceUnit *int    `json:"sizePerResourceUnit,omitempty"`
+	Resource            *string `json:"resource,omitempty"`
+
+	forceSendFields []string
+	nullFields      []string
+}
+
+// region BlockDeviceMapping
+
+func (o ClusterBlockDeviceMapping) MarshalJSON() ([]byte, error) {
+	type noMethod ClusterBlockDeviceMapping
+	raw := noMethod(o)
+	return jsonutil.MarshalJSON(raw, o.forceSendFields, o.nullFields)
+}
+
+func (o *ClusterBlockDeviceMapping) SetDeviceName(v *string) *ClusterBlockDeviceMapping {
+	if o.DeviceName = v; o.DeviceName == nil {
+		o.nullFields = append(o.nullFields, "DeviceName")
+	}
+	return o
+}
+
+func (o *ClusterBlockDeviceMapping) SetNoDevice(v *string) *ClusterBlockDeviceMapping {
+	if o.NoDevice = v; o.NoDevice == nil {
+		o.nullFields = append(o.nullFields, "NoDevice")
+	}
+	return o
+}
+
+func (o *ClusterBlockDeviceMapping) SetVirtualName(v *string) *ClusterBlockDeviceMapping {
```

Comment:
recheck the documentation for this param

Reply:
removed unused parameters VirtualName.